pandas/core/dtypes/missing.py

```diff
@@ -187,7 +187,10 @@ def _use_inf_as_na(key):
 
 
 def _isna_ndarraylike(obj):
-    values = getattr(obj, 'values', obj)
+    if not is_extension_array_dtype(obj):
+        values = getattr(obj, 'values', obj)
+    else:
+        values = obj
```

Comment:
OK, the above was actually only needed to get my experiment passing (where I raised in the `PeriodArray.values` attribute, to detect where it was used). 
But of course, the below uses the original `obj` again, so this is not an issue (still, a bit complex logic to follow though)

Reply:
Would you recommend changing this on the PR now? I think changing it is likely to cause confusion.

Ahh, but we don't want `.values` here so we should change it...

Reply:
Well, we get the `.values`, but then when the `obj` was EA, we don't do anything with it. So I think we can leave it as is, for now (although it is a bit ugly).